libs/sdk/skyux-stylelint/src/rules/no-ng-deep.test.ts

```diff
@@ -0,0 +1,240 @@
+      },
+      {
+        code: '.ng-deep-like-class { color: red; }',
+        description: 'classes containing "ng-deep" should be allowed',
+      },
+      {
+        code: '[data-ng-deep] { color: red; }',
+        description: 'attribute selectors with ng-deep should be allowed',
+      },
+      {
+        code: '/* ::ng-deep would be invalid */',
+        description: 'comments containing ::ng-deep should be allowed',
+      },
+      {
+        code: '@media (min-width: 768px) { .my-class { color: red; } }',
+        description: 'media queries without ng-deep should be allowed',
+      },
+      {
+        code: '.deep { color: blue; }',
+        description: 'classes with "deep" should be allowed',
+      },
+      {
+        code: '.ng-container { display: contents; }',
+        description: 'other ng- prefixed classes should be allowed',
+      },
+    ],
+
+    reject: [
+      {
+        code: '::ng-deep .my-class { color: red; }',
+        description: 'standard ::ng-deep usage should be rejected',
+        message:
+          'Unexpected usage of "::ng-deep". The Angular team strongly discourages new use of ::ng-deep as it breaks component encapsulation. Consider using :host or component communication patterns instead.',
+        line: 1,
+      },
+      {
+        code: '.parent ::ng-deep .child { margin: 10px; }',
+        description: '::ng-deep in descendant selectors should be rejected',
+        message:
+          'Unexpected usage of "::ng-deep". The Angular team strongly discourages new use of ::ng-deep as it breaks component encapsulation. Consider using :host or component communication patterns instead.',
+        line: 1,
+      },
+      {
+        code: ':host ::ng-deep .global { color: blue; }',
+        description: '::ng-deep with :host should be rejected',
+        message:
+          'Unexpected usage of "::ng-deep". The Angular team strongly discourages new use of ::ng-deep as it breaks component encapsulation. Consider using :host or component communication patterns instead.',
+        line: 1,
+      },
+      {
+        code: '::NG-DEEP .uppercase { color: green; }',
+        description: 'uppercase ::NG-DEEP should be rejected',
+        message:
+          'Unexpected usage of "::ng-deep". The Angular team strongly discourages new use of ::ng-deep as it breaks component encapsulation. Consider using :host or component communication patterns instead.',
+        line: 1,
+      },
+      {
+        code: ':: ng-deep .with-space { color: purple; }',
+        description: ':: ng-deep with space should be rejected',
+        message:
+          'Unexpected usage of "::ng-deep". The Angular team strongly discourages new use of ::ng-deep as it breaks component encapsulation. Consider using :host or component communication patterns instead.',
+        line: 1,
+      },
+      {
+        code: ': ng-deep .single-colon { color: orange; }',
```

Comment:
This is a valid test - but did we intend to have a space here? Should we have a test with the space and one without for this?

Reply:
Hm, on second thought, the Angular compiler only recognizes `::ng-deep`, so I'll simplify the rule and tests to accommodate.